db/Pos/DB/Functions.hs

```diff
@@ -59,14 +87,17 @@ dbDecode =
     onParseError :: ByteString -> Maybe ByteString -> Text -> m a
     onParseError rawKey rawValMaybe errMsg =
         let valueBuilder = maybe "" (bprint (", value = " %shown)) rawValMaybe
-        in throwM $ DBMalformed $ sformat fmt rawKey valueBuilder errMsg
-    fmt =
+        in throwM $ DBMalformed $ sformat fmtMalformed rawKey valueBuilder errMsg
+    fmtMalformed =
         "A key or value stored in DB is malformed, key = "%shown%
         builder%
         ", err: "%stext
 
-dbDecodeMaybe :: (Bi v) => ByteString -> Maybe v
-dbDecodeMaybe = rightToMaybe . decodeFull
+dbDecodeMaybe :: forall v . (Bi v) => ByteString -> Maybe v
+dbDecodeMaybe bs =
```

Comment:
It's also used in `dbDecodeMaybeWP` which decodes keys. So it shouldn't always decode `(Word8, v)`. Some refactoring is needed. Also it shouldn't be exported, because it's not used anywhere.